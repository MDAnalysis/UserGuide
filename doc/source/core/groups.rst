.. -*- coding: utf-8 -*-

====================
Groups
====================


For convenience, chemically meaningful groups of Atoms such as a 
Residue or a Segment (typically a whole molecule or all of the solvent) 
also exist as containers, as well as groups of these units (
ResidueGroup, SegmentGroup).

AtomGroups
==========
AtomGroup instances can be easily created 
(e.g., from an AtomGroup.select_atoms() selection or simply by slicing).

