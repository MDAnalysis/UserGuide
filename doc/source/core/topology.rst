.. -*- coding: utf-8 -*-

====================
Topology
====================

MDA Topology =/= 'force field' topology. Connectivity only. See ParmEd etc for other stuff.

Adding attributes to Topology
=============================

stuff