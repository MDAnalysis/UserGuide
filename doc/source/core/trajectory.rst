.. -*- coding: utf-8 -*-

====================
Trajectory
====================

Not intuitive.

.positions() updates

Auxiliary trajectory readers
============================

