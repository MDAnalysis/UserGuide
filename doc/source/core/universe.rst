.. -*- coding: utf-8 -*-

====================
Universes
====================

| *If you wish to make an apple pie from scratch, you must first invent the universe.*
| - Carl Sagan, Cosmos

| *In the beginning the Universe was created.*
| *This has made a lot of people very angry and been widely regarded as a bad move.*
| â€• Douglas Adams, The Restaurant at the End of the Universe

The Universe class ties a topology and a trajectory together. 
Almost all code in MDAnalysis starts with a Universe.

Creating a Universe
===================

Loading from files
------------------

.. code-block:: python

u = Universe(topology, trajectory)          # read system from file(s)
u = Universe(pdbfile)                       # read atoms and coordinates from PDB or GRO
u = Universe(topology, [traj1, traj2, ...]) # read from a list of trajectories
u = Universe(topology, traj1, traj2, ...)   # read from multiple trajectories

Many words here

Constructing from AtomGroups
----------------------------

Merge()

