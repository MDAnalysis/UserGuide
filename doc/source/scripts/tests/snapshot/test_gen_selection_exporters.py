from gen_selection_exporters import SelectionExporterWriter
from MDAnalysis.core import selection as sel
from unittest.mock import patch

def test_SelectionExporterWriter(snapshot):
    with patch("builtins.open"):
        se = SelectionExporterWriter()
    assert se.lines == snapshot
