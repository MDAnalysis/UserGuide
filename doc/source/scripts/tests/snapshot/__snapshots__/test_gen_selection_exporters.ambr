# serializer version: 1
# name: test_SelectionExporterWriter
  list([
    list([
      '`CHARMM`_',
      'str',
      'CHARMM selection of individual atoms',
      ':class:`MDAnalysis.selections.charmm.SelectionWriter`',
    ]),
    list([
      '`Gromacs`_',
      'ndx',
      'GROMACS index file',
      ':class:`MDAnalysis.selections.gromacs.SelectionWriter`',
    ]),
    list([
      '`Jmol`_',
      'spt',
      'Jmol selection commands',
      ':class:`MDAnalysis.selections.jmol.SelectionWriter`',
    ]),
    list([
      '`PyMol`_',
      'pml',
      'PyMOL selection string',
      ':class:`MDAnalysis.selections.pymol.SelectionWriter`',
    ]),
    list([
      '`VMD`_',
      'vmd',
      'VMD macros, available in Representations',
      ':class:`MDAnalysis.selections.vmd.SelectionWriter`',
    ]),
  ])
# ---
